From c2428419e7d9596f07b62269954ea9c4fcc7e422 Mon Sep 17 00:00:00 2001
From: Frank Wang <frank.wang@rock-chips.com>
Date: Fri, 16 Dec 2016 08:38:11 +0800
Subject: [PATCH] arm64: dts: rockchip: keep vbus always on for usb2.0 host of
 rk3399

One basic condition of usb remote wakeup is vbus on, so we add
regulator-always-on property for vcc5v0_host regulator in this adds.

In the previous codes, ehci/ohci-platform did not power off
vcc5v0_host regulator due to some oddish codes, and we have fixed it
at commit b5a0a9e8794d.

Change-Id: I95c225c9c3aeec6e346d62e61fdcde5e5e02d143
Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts      | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-box.dtsi                 | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi                 | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts      | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi            | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts             | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts           | 1 +
 8 files changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts
index dd68b23cc642..877d9f4ac0ef 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box-808-android.dts
@@ -90,6 +90,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	vdd_center: vdd-center {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
index 9014d1c29b26..3a2f5255a119 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
@@ -87,6 +87,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	vdd_log: vdd-log {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
index 696004f351b3..759babc4d468 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
@@ -63,6 +63,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	backlight: backlight {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts
index 477980f01a7f..af63fadfb594 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android-next.dts
@@ -97,6 +97,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	vdd_log: vdd-log {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
index 8af5d80db980..cbf8286b48d9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
@@ -85,6 +85,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	vdd_log: vdd-log {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
index 2d3bb78576fc..000d8a5df920 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
@@ -147,6 +147,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	vcc5v0_sys: vcc5v0-sys {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
index e59b5984cdfc..1cb199edeb59 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
@@ -89,6 +89,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	vdd_log: vdd-log {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts
index 09329e9031af..4472baa0115a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-vr-android.dts
@@ -86,6 +86,7 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&host_vbus_drv>;
 		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	backlight: backlight {
-- 
2.35.3

